client_internals_it_test.go

```diff
@@ -149,3 +152,53 @@ func (h *riggedInvocationHandler) Invoke(inv invocation.Invocation) (int64, erro
 	}
 	return h.Handler.Invoke(inv)
 }
+
+func TestClusterID(t *testing.T) {
+	it.SkipIf(t, "oss")
+	clientTester(t, func(t *testing.T, smart bool) {
+		ctx := context.Background()
+		cls1 := it.StartNewClusterWithOptions("clusterId-test-cluster1", 15701, it.MemberCount())
```

Comment:
Can we register a defer call for cls1.Shutdown()? I am not sure if this has any side effects if cluster is already closed, however there are multiple code paths until we reach line
```cls1.Shutdown()``` and it would be great that we do not leak this cluster if something goes wrong.

Reply:
You are right, thank you. Fixed at [49d4f5b](https://github.com/hazelcast/hazelcast-go-client/pull/739/commits/49d4f5ba7721a379b2db599d9d0df6c6c7e4f4b5).